### PR TITLE
PAYARA-2541 synch HK2 Dom class race condition

### PR DESCRIPTION
**Pull request by lprimak**
synchronized all getter access to the Dom class due to Dom not being thread-safe, as per PAYARA-2541

### CODE REVIEW

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

File: nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/NetworkListener.java [hunk 1]
```diff
@@ -226,9 +226,16 @@ public static ThreadPool findThreadPool(NetworkListener listener) {
             List<ThreadPool> list = listeners.getThreadPool();
```
Comment (MattGill98) [inline nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/NetworkListener.java]: Needs copyright header.

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
jenkins test

**Pull request merged**